fuse/lag_node.go

```diff
@@ -0,0 +1,74 @@
+package fuse
+
+import (
+	"context"
+	"strconv"
+	"syscall"
+
+	"bazil.org/fuse"
+	"bazil.org/fuse/fs"
+)
+
+const LagFilename = ".lag"
+
+var _ fs.Node = (*LagNode)(nil)
+var _ fs.NodeForgetter = (*LagNode)(nil)
+var _ fs.HandleReadAller = (*LagNode)(nil)
+var _ fs.NodeListxattrer = (*PosNode)(nil)
+var _ fs.NodeGetxattrer = (*PosNode)(nil)
+var _ fs.NodeSetxattrer = (*PosNode)(nil)
+var _ fs.NodeRemovexattrer = (*PosNode)(nil)
+var _ fs.NodePoller = (*PosNode)(nil)
+
+type LagNode struct {
+	fsys *FileSystem
+}
+
+func newLagNode(fsys *FileSystem) *LagNode {
+	return &LagNode{fsys: fsys}
+}
+
+func (n *LagNode) Attr(ctx context.Context, attr *fuse.Attr) error {
+	attr.Mode = 0o444
+	attr.Size = uint64(len(n.content()))
+	attr.Uid = uint32(n.fsys.Uid)
+	attr.Gid = uint32(n.fsys.Gid)
+	return nil
+}
+
+func (n *LagNode) Forget() { n.fsys.root.ForgetNode(n) }
+
+func (n *LagNode) ReadAll(ctx context.Context) ([]byte, error) {
+	return n.content(), nil
+}
+
+func (n *LagNode) content() []byte {
+	if ts := n.fsys.store.PrimaryTimestamp(); ts > 0 {
+		return []byte(strconv.FormatInt(ts, 10))
+	}
+	return []byte("-1")
+}
```

Comment:
A couple questions:
1. Does this need to be a separate function? Can it just go into `ReadAll()`?
2. Since it's the "lag" node, should we return the lag (difference between time.Now() and `PrimaryTimestamp()`)? If a user is calculating the lag after reading this file then they're also incurring the overhead of the FUSE interactions (which should be small).
3. I typically add a newline after the value so people can `cat` the file and read it easily.

Reply:
1. I abstracted it, since I'm using it to calculate size too. 
2. Up to you. I did it this way to make clock skew the user's problem instead of ours, but giving them the delta is definitely more user friendly. 
3. 👍